www/js/app.js

```diff
@@ -939,7 +939,9 @@ define(['jquery', 'zimArchiveLoader', 'util', 'uiUtil', 'cookies','abstractFiles
         var listLength = dirEntryArray.length < MAX_SEARCH_RESULT_SIZE ? dirEntryArray.length : MAX_SEARCH_RESULT_SIZE;
         for (var i = 0; i < listLength; i++) {
             var dirEntry = dirEntryArray[i];
-            articleListDivHtml += '<a href="#" dirEntryId="' + dirEntry.toStringId().replace(/'/g, '&apos;') +
+            var dirEntryStringId = dirEntry.toStringId();
+            dirEntryStringId = uiUtil.htmlEscapeChars(dirEntryStringId);
```

Comment:
Very minor : these 2 lines could be combined into a single one, so that the `dirEntryStringId` can not have two different values (escaped and unescaped) : 
```suggestion
            var dirEntryStringId = uiUtil.htmlEscapeChars(dirEntry.toStringId());
```